bpf/lib/fib.h

```diff
@@ -47,20 +47,26 @@ redirect_direct_v6(struct __ctx_buff *ctx __maybe_unused,
 		return CTX_ACT_DROP;
 	}
 
-	oif = fib_params.ifindex;
+	*oif = fib_params.ifindex;
+# else
+	*oif = DIRECT_ROUTING_DEV_IFINDEX;
 # endif /* ENABLE_SKIP_FIB */
 
 	ret = ipv6_l3(ctx, l3_off, NULL, NULL, METRIC_EGRESS);
 	if (unlikely(ret != CTX_ACT_OK))
 		return ret;
-	if (no_neigh)
-		return redirect_neigh(oif, nh, nh ? sizeof(*nh) : 0, 0);
+	if (no_neigh) {
+		if (nh)
+			return redirect_neigh(*oif, nh, nh ? sizeof(*nh) : 0, 0);
```

Comment:
nit: `nh ? sizeof(*nh) : 0` nh is known to be non-zero at this point

Reply:
See also my commit from last week: https://github.com/cilium/cilium/pull/20659

Reply:
Silly mistake on my end :facepalm:. Fixed it, the two PRs are now the same, except that the commit description in this one is more complete (more explanations about the cause of the issue we fix, and a note on the troubles with the verifier).